### ISSUE TITLE
Upgrading to go1.15 results in a breaking error

### ISSUE THREAD

**Issue by rkuprov** [testcontainers/testcontainers-go#232]
**Describe the bug**
Running the package after upgrading to go1.15 causes failure in container creation with this error:
`failed to create container: creating reaper failed: could not start container: host port waiting failed: error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.40/exec/4518d0eb0fbb7a35e6917e015ce8300646d703ad809453be8f500b64113f25a9/start": net/http: HTTP/1.x transport connection broken: unsupported transfer encoding: "identity"`

**To Reproduce**
Not sure that I can.

**Expected behavior**
Not to have this error

** docker info **
output of the command:

```
$ docker info
```

**Additional context**

**Comment by sazzer** [comment]
Getting this exact same error here. Works on 1.14.7 and fails on 1.15.

In case it helps:
```
coxg@coxgmbpsh [11:23:37] [~/source/me/mire/service] [go *]
-> % ggovm install 1.14.7

coxg@coxgmbpsh [11:23:48] [~/source/me/mire/service] [go *]
-> % go test ./integration/health
ok  	github.com/sazzer/mire/service/integration/health	5.816s

coxg@coxgmbpsh [11:24:11] [~/source/me/mire/service] [go *]
-> % ggovm install 1.15

coxg@coxgmbpsh [11:25:10] [~/source/me/mire/service] [go *]
-> % go test ./integration/health
{"level":"info","time":"2020-08-16T11:25:17+01:00","message":"Starting database"}
2020/08/16 11:25:17 Starting container id: 649bbf85fe25 image: quay.io/testcontainers/ryuk:0.2.3
2020/08/16 11:25:18 Waiting for container id 649bbf85fe25 image: quay.io/testcontainers/ryuk:0.2.3
{"level":"debug","time":"2020-08-16T11:25:18+01:00","message":"Building Mire..."}
{"level":"info","name":"database","time":"2020-08-16T11:25:18+01:00","message":"Component health"}
2020/08/16 11:25:18 [coxgmbpsh.local/30Brx8Ak3a-000001] "GET http://example.com/health HTTP/1.1" from 192.0.2.1:1234 - 200 60B in 579.679µs
{"level":"info","time":"2020-08-16T11:25:18+01:00","message":"Stopping database"}
--- FAIL: TestHealth (0.40s)
    database.go:31:
        	Error Trace:	database.go:31
        	            				service.go:18
        	            				health_test.go:14
        	Error:      	Received unexpected error:
        	            	error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.40/exec/4798a0fc783bc5e25947c792a4cc2d6c4c9fdffd49970abf8dd9b6a69ca659c1/start": net/http: HTTP/1.x transport connection broken: unsupported transfer encoding: "identity"
        	            	host port waiting failed
        	            	github.com/testcontainers/testcontainers-go/wait.(*HostPortStrategy).WaitUntilReady
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/wait/host_port.go:97
        	            	github.com/testcontainers/testcontainers-go.(*DockerContainer).Start
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/docker.go:160
        	            	github.com/testcontainers/testcontainers-go.(*DockerProvider).RunContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/docker.go:642
        	            	github.com/testcontainers/testcontainers-go.NewReaper
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/reaper.go:73
        	            	github.com/testcontainers/testcontainers-go.(*DockerProvider).CreateContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/docker.go:479
        	            	github.com/testcontainers/testcontainers-go.GenericContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/generic.go:43
        	            	github.com/sazzer/mire/service/integration.NewDatabase
        	            		/Users/coxg/source/me/mire/service/integration/database.go:27
        	            	github.com/sazzer/mire/service/integration.NewService
        	            		/Users/coxg/source/me/mire/service/integration/service.go:18
        	            	github.com/sazzer/mire/service/integration/health_test.TestHealth
        	            		/Users/coxg/source/me/mire/service/integration/health/health_test.go:14
        	            	testing.tRunner
        	            		/Users/coxg/.go/src/testing/testing.go:1108
        	            	runtime.goexit
        	            		/Users/coxg/.go/src/runtime/asm_amd64.s:1374
        	            	could not start container
        	            	github.com/testcontainers/testcontainers-go.(*DockerProvider).RunContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/docker.go:643
        	            	github.com/testcontainers/testcontainers-go.NewReaper
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/reaper.go:73
        	            	github.com/testcontainers/testcontainers-go.(*DockerProvider).CreateContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/docker.go:479
        	            	github.com/testcontainers/testcontainers-go.GenericContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/generic.go:43
        	            	github.com/sazzer/mire/service/integration.NewDatabase
        	            		/Users/coxg/source/me/mire/service/integration/database.go:27
        	            	github.com/sazzer/mire/service/integration.NewService
        	            		/Users/coxg/source/me/mire/service/integration/service.go:18
        	            	github.com/sazzer/mire/service/integration/health_test.TestHealth
        	            		/Users/coxg/source/me/mire/service/integration/health/health_test.go:14
        	            	testing.tRunner
        	            		/Users/coxg/.go/src/testing/testing.go:1108
        	            	runtime.goexit
        	            		/Users/coxg/.go/src/runtime/asm_amd64.s:1374
        	            	creating reaper failed
        	            	github.com/testcontainers/testcontainers-go.(*DockerProvider).CreateContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/docker.go:481
        	            	github.com/testcontainers/testcontainers-go.GenericContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/generic.go:43
        	            	github.com/sazzer/mire/service/integration.NewDatabase
        	            		/Users/coxg/source/me/mire/service/integration/database.go:27
        	            	github.com/sazzer/mire/service/integration.NewService
        	            		/Users/coxg/source/me/mire/service/integration/service.go:18
        	            	github.com/sazzer/mire/service/integration/health_test.TestHealth
        	            		/Users/coxg/source/me/mire/service/integration/health/health_test.go:14
        	            	testing.tRunner
        	            		/Users/coxg/.go/src/testing/testing.go:1108
        	            	runtime.goexit
        	            		/Users/coxg/.go/src/runtime/asm_amd64.s:1374
        	            	failed to create container
        	            	github.com/testcontainers/testcontainers-go.GenericContainer
        	            		/Users/coxg/go/pkg/mod/github.com/testcontainers/testcontainers-go@v0.7.0/generic.go:45
        	            	github.com/sazzer/mire/service/integration.NewDatabase
        	            		/Users/coxg/source/me/mire/service/integration/database.go:27
        	            	github.com/sazzer/mire/service/integration.NewService
        	            		/Users/coxg/source/me/mire/service/integration/service.go:18
        	            	github.com/sazzer/mire/service/integration/health_test.TestHealth
        	            		/Users/coxg/source/me/mire/service/integration/health/health_test.go:14
        	            	testing.tRunner
        	            		/Users/coxg/.go/src/testing/testing.go:1108
        	            	runtime.goexit
        	            		/Users/coxg/.go/src/runtime/asm_amd64.s:1374
        	Test:       	TestHealth
FAIL
FAIL	github.com/sazzer/mire/service/integration/health	0.629s
FAIL
```

Cheers

**Comment by rkuprov** [comment]
@sazzer , thank you! I felt a bit silly posting the  initial report.

**Comment by sazzer** [comment]
No worries. I'm glad you did because I thought it was just my computer at
first!

**Comment by kostyay** [comment]
Seems to be related to this: https://github.com/golang/go/issues/40735

**Comment by iulian-gm** [comment]
Is there a workaround for this?

**Comment by rkuprov** [comment]
I just had to downgrade to 1.14.7. Dave Cheney wants a way to reproduce the issue. I'm not sure why, since the code provided by h8liu in the issue you (@iulian-gm ) posted already does that.

**Comment by mdelapenya** [comment]
Hey, we are adding coverage for different versions of Go in the test suite on Travis (https://github.com/testcontainers/testcontainers-go/pull/233)

Once merged, could you please add a test reproducing the issue? I think it would be easy with the scenario proposed by @sazzer above.

Thanks!

**Comment by lujjjh** [comment]
My workaround:

```go
import (
	"context"
	"net"
	"os"

	"github.com/testcontainers/testcontainers-go"
	"github.com/valyala/fasthttp"
)

func GenericContainer(ctx context.Context, req testcontainers.GenericContainerRequest) (testcontainers.Container, error) {
	ln, err := net.Listen("tcp", "127.0.0.1:0")
	if err != nil {
		return nil, err
	}
	client := &fasthttp.Client{
		Dial: func(string) (net.Conn, error) {
			return net.Dial("unix", "/var/run/docker.sock")
		},
	}
	go fasthttp.Serve(ln, func(ctx *fasthttp.RequestCtx) {
		_ = client.Do(&ctx.Request, &ctx.Response)
	})
	const kDockerHost = "DOCKER_HOST"
	previousDockerHost, ok := os.LookupEnv(kDockerHost)
	os.Setenv(kDockerHost, "http://"+ln.Addr().String())
	defer func() {
		if ok {
			os.Setenv(kDockerHost, previousDockerHost)
		} else {
			os.Unsetenv(kDockerHost)
		}
	}()
	return testcontainers.GenericContainer(ctx, req)
}
```

**Comment by insta-code1** [comment]
Note: I recently upgraded my mac os and encountered this error too. Upgrading the docker desktop application fixed the issue.

**Comment by dmarkhas** [comment]
Can confirm this is fixed in Docker Desktop 2.5.0.1 for MacOS

**Comment by rkuprov** [comment]
My bad. Should've closed it a while ago!
